examples/basic_candlepin.pp

```diff
@@ -27,9 +29,9 @@
   notify  => Service['tomcat'],
 } ->
 package { ['java']: } ->
-exec { 'add ca cert':
-  command => "/usr/bin/keytool -import -trustcacerts -v -keystore /etc/candlepin/certs/keystore -storepass ${keystore_password} -alias candlepin-ca -file /etc/candlepin/certs/candlepin-ca.crt -noprompt",
-  unless  => "/usr/bin/keytool -list -keystore ${keystore} -storepass ${keystore_password} -alias candlepin-ca | grep $(openssl x509 -noout -fingerprint -in ${ca_cert} | cut -d '=' -f 2)",
+exec { 'Create truststore':
```

Comment:
With Puppet 6 we could use Deferred and implement a function, but sadly we still need to support Puppet 5.

https://puppet.com/blog/agent-side-functions-in-puppet-6/ shows it could be implemented as:
```puppet
file { $truststore:
  ensure  => file,
  content => Deferred('pem2pkcs12', [$ca_cert, $truststore_password]),
  mode    => '0440',
  owner   => 'root',
  group   => 'tomcat',
}
```

Of course it would require a `pem2pkcs12` function.

Reply:
Why do need that for the *example* ?

Reply:
Because I should look at filenames. I assumed this was part of the module itself, but it's really in puppet-certs.